wget/wget.yaml

```diff
@@ -0,0 +1,30 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: wget
+spec:
+  params:
+   - name: url
+     description: The url we want to download file from
+     type: string
+     default: ""
+   - name: diroptions
+     description: The directory path we want to save file to
+     type: array
+     default: []
+   - name: options
+     description: The arguments to pass to wget
+     type: array
+     default: []
+   - name: filename
+     description: The filename we want to change our file to
+     type: string
+     default: ""
+  workspaces:
+   - name: wget-workspace
+     description: The folder where we write the wget'ed file to  
+  steps:
+    - name: wget
+      image: mwendler/wget
```

Comment:
Sure

Reply:
Bear in mind that the wget shipped with alpine is a minimal wget builtin to busybox : 

![image](https://user-images.githubusercontent.com/98980/84378387-61355600-abe4-11ea-901c-1cf99ad8c5db.png)

so perhaps not something the user would expect....

Reply:
Hmm, good point. We could create a (re)container with like the original one? Maybe put a newer version in the tecktoncd namespace?

(might be a good case study of using tekton paying attention a repository and deploying a new container automagicaly)

Reply:
How about for now we add image url as param, so users can specify any versions in taskrun?

Reply:
yeah, I don't think we have a good story around hosting the images for catalog yet. To me, using a default image and letting the user override sounds reasonable.

Reply:
What about we are using this image as default not ideal because the image purpose is for a different thing and may be slighty bigger but it does have the wget official binary and it's an official image from docker.inc : 


```
[root@openshift2 fedora]# docker run -it buildpack-deps
root@927c2eb6149c:/# wget --version
GNU Wget 1.20.1 built on linux-gnu.

-cares +digest -gpgme +https +ipv6 +iri +large-file -metalink +nls
+ntlm +opie +psl +ssl/gnutls

Wgetrc:
    /etc/wgetrc (system)
Locale:
    /usr/share/locale
Compile:
    gcc -DHAVE_CONFIG_H -DSYSTEM_WGETRC="/etc/wgetrc"
    -DLOCALEDIR="/usr/share/locale" -I. -I../../src -I../lib
    -I../../lib -Wdate-time -D_FORTIFY_SOURCE=2
    -I/usr/include/p11-kit-1 -DHAVE_LIBGNUTLS -DNDEBUG -g -O2
    -fdebug-prefix-map=/build/wget-EiPT9d/wget-1.20.1=.
    -fstack-protector-strong -Wformat -Werror=format-security
    -DNO_SSLv2 -D_FILE_OFFSET_BITS=64 -g -Wall
Link:
    gcc -I/usr/include/p11-kit-1 -DHAVE_LIBGNUTLS -DNDEBUG -g -O2
    -fdebug-prefix-map=/build/wget-EiPT9d/wget-1.20.1=.
    -fstack-protector-strong -Wformat -Werror=format-security
    -DNO_SSLv2 -D_FILE_OFFSET_BITS=64 -g -Wall -Wl,-z,relro -Wl,-z,now
    -lpcre2-8 -luuid -lidn2 -lnettle -lgnutls -lz -lpsl ftp-opie.o
    gnutls.o http-ntlm.o ../lib/libgnu.a

Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later
<http://www.gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Originally written by Hrvoje Niksic <hniksic@xemacs.org>.
Please send bug reports and questions to <bug-wget@gnu.org>.
```

Reply:
Honest question; We have `https://hub.docker.com/u/tektoncd` couldn't we just create a `tektoncd/wget` alpine image and host it there?

Reply:
Nice! Yeah using a pinned version of `buildpack-deps` sounds good to me. Looks like they specifically have a `curl` tag that only contains `wget` and `curl`.:
```
curl
This variant includes just the curl, wget, and ca-certificates packages. This is perfect for cases like the Java JRE, where downloading JARs is very common and necessary, but checking out code isn't.
```

@jjasghar We could host the image too but then we'd have to figure out a way to keep it updated on a regular basis. I'd lean towards using an already maintained image if possible!

Reply:
> Honest question; We have https://hub.docker.com/u/tektoncd couldn't we just create a tektoncd/wget alpine image and host it there?

Yes we could, in the plumbing we have already a couple of them cc @afrittoli 

but we really need a policy/story around images and how we want to manage them/generate them and if we want to provide/support them (as an organisation)

(or after reading @dibyom 's comment that's exactly what i meant)